test/e2e/fleet_test.go

```diff
@@ -123,6 +123,114 @@ func TestFleetStrategyValidation(t *testing.T) {
 	verifyErr(err)
 }
 
+func TestFleetScaleUpAllocateEditAndScaleDownToZero(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+
+	client := framework.AgonesClient.AgonesV1()
+
+	flt := defaultFleet(framework.Namespace)
+	flt.Spec.Replicas = 1
+	// if PlayerTracking is enabled, apply InitialCapacity
+	if runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		flt.Spec.Template.Spec.Players = &agonesv1.PlayersSpec{
+			InitialCapacity: 0,
+		}
+	}
+	flt, err := client.Fleets(framework.Namespace).Create(ctx, flt, metav1.CreateOptions{})
+	if assert.Nil(t, err) {
+		defer client.Fleets(framework.Namespace).Delete(ctx, flt.ObjectMeta.Name, metav1.DeleteOptions{}) // nolint:errcheck
+	}
+
+	assert.Equal(t, int32(1), flt.Spec.Replicas)
+
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(flt.Spec.Replicas))
+
+	// Scale up to 5 replicas
+	const targetScale = 5
+	flt = scaleFleetPatch(ctx, t, flt, targetScale)
+	framework.AssertFleetCondition(t, flt, e2e.FleetReadyCount(targetScale))
+
+	// Allocate 1 replica
+	gsa := framework.CreateAndApplyAllocation(t, flt)
+
+	framework.AssertFleetCondition(t, flt, func(log *logrus.Entry, fleet *agonesv1.Fleet) bool {
+		return fleet.Status.AllocatedReplicas == 1
+	})
+
+	flt, err = client.Fleets(framework.Namespace).Get(ctx, flt.ObjectMeta.GetName(), metav1.GetOptions{})
+	assert.Nil(t, err)
+
+	// Edit PlayersInitialCapacity to 4, if PlayerTracking is enabled
+	// Change Players Initial Capacity to trigger creating a new GSSet
+	// Or change port, if PlayerTracking is not enabled
+	fltCopy := flt.DeepCopy()
+	if runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		fltCopy.Spec.Template.Spec.Players.InitialCapacity = 4
+	} else {
+		fltCopy.Spec.Template.Spec.Ports = []agonesv1.GameServerPort{{
+			ContainerPort: 6000,
+			Name:          "gameport",
+			PortPolicy:    agonesv1.Dynamic,
+			Protocol:      corev1.ProtocolUDP,
+		}}
+	}
```

Comment:
Oh this works! I was going to suggest adding a label, but changing a port works perfectly as well. No issues with the approach on my end!

```suggestion
fltCopy.Spec.Template.Spec.Ports = []agonesv1.GameServerPort{{
	ContainerPort: 6000,
	Name:          "gameport",
	PortPolicy:    agonesv1.Dynamic,
	Protocol:      corev1.ProtocolUDP,
}}
```

What I would then suggest, just get rid of all the player tracking and feature flag parts here and elsewhere in this e2e test, and just to the port manipulation -- there's nothing inherent to player tracking in this ticket, it was just  way to trigger a rolling update.

It will be much easier to read 👍🏻

Reply:
Thank you!! I updated the codebase.